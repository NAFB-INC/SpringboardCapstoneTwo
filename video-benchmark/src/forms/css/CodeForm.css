.form-text{
    font-size:150%;
    font-weight:bold;
    margin-bottom:5%;
}

.submit-btn {
	box-shadow: 3px 4px 0px 0px #899599;
	background:linear-gradient(to bottom, #D5B9B2 5%, #e2d3e2 100%);
	background-color:#D5B9B2;
	border-radius:15px;
	border:1px solid #d6bcd6;
	display:inline-block;
	cursor:pointer;
	color:#6D2E46;
	font-family:Arial;
	font-size:130%;
	padding:7px 25px;
	text-decoration:none;
    text-shadow:0px 1px 0px #e5d7d3;
    margin-top:2%;
}
.submit-btn:hover {
	background:linear-gradient(to bottom, #bab1ba 5%, #ededed 100%);
	background-color:#bab1ba;
}
.submit-btn:active {
	position:relative;
	top:1px;
}