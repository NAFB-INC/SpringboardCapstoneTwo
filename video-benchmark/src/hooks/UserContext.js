import React from "react";

//storage place for app-wide data
const UserContext = React.createContext();

export default UserContext;
