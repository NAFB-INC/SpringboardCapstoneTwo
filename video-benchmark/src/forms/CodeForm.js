import React,{useContext,useState} from "react";
import useFields from "../hooks/useFields";
import UserContext from "../hooks/UserContext";
import VideoAPI from "../api/VideoAPI";

function AudienceCodeForm({ setCode, presenter }) {
    const { setStage } = useContext(UserContext);
    const [errors,setErrors] = useState([]);
    const [formData, handleChange, resetForm] = useFields({
        audience_code: '',
        presenter_pass: ''
    })


    const handleSubmit = e => {
        e.preventDefault();

        async function checkCode(){
            return (await VideoAPI.findIfValidCode(formData.audience_code,formData.presenter_pass));
        }

        let valid;
        checkCode().then((tOrF)=>{
            valid=tOrF;

            if(presenter){
                if(valid){
                    setCode(formData.audience_code);
                    setStage(3);
                }else{
                    setErrors(["Code not found!"])
                }
            }else {
                if(valid){
                    setCode(formData.audience_code);
                    setStage(3);
                }else{
                    setErrors(["Code not found!"])
                }
            }
        });

        resetForm();
    }

    if(presenter){
        return (
            <div className="AudienceCodeForm">
                Have a presentation code and password?
                Please enter it here!
                <form onSubmit={handleSubmit}>
                    <label htmlFor="audience_code">Code: </label>
                    <input autoComplete="audience-code" name="audience_code" type="text" placeholder="abc123" value={formData.audience_code} onChange={handleChange}></input>
                    <label htmlFor="presenter_pass">Password: </label>
                    <input autoComplete="presenter-password" name="presenter_pass" type="password" placeholder="xyzpdq" value={formData.presenter_pass} onChange={handleChange}></input>
                    <div>
                        {errors.map((e)=>(
                            <li className="error" key={`error${e}`}>{e}</li>
                        ))}
                    </div>
                    <button>Submit</button>
                </form>
            </div>
        );
    }else{
        return (
            <div className="AudienceCodeForm">
                Have a presentation code?
                Please enter it here!
                <form onSubmit={handleSubmit}>
                    <label htmlFor="audience_code">Code: </label>
                    <input name="audience_code" type="text" placeholder="abc123" value={formData.audience_code} onChange={handleChange}></input>
                    <div>
                        {errors.map((e)=>(
                            <li className="error" key={`error${e}`}>{e}</li>
                        ))}
                    </div>
                    <button>Submit</button>
                </form>
            </div>
        );
    }
}

export default AudienceCodeForm;