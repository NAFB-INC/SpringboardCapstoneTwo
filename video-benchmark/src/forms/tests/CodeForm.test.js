import React from "react";
import { render } from '@testing-library/react';
import CodeForm from '../CodeForm';
import UserContext from "../../hooks/UserContext";

let setStage = function(){};
const testCase1 = (
  <div>
      <UserContext.Provider value={{ setStage }}>
          <CodeForm />
      </UserContext.Provider>
  </div>);

it("renders without crashing", function() {
  render(testCase1);
});

it("matches snapshot", function() {
  const { asFragment } = render(testCase1);
  expect(asFragment()).toMatchSnapshot();
});