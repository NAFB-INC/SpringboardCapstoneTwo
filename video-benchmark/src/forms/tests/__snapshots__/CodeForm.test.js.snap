// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="AudienceCodeForm"
    >
      <div
        class="form-text"
      >
        Have a presentation code? Please enter it here!
      </div>
      <form>
        <div
          class="form-box"
        >
          <label
            for="audience_code"
          >
            Code: 
          </label>
          <input
            name="audience_code"
            placeholder="abc123"
            type="text"
            value=""
          />
        </div>
        <div />
        <button
          class="submit-btn"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
